### DISCUSSION TITLE
No show of import

### DISCUSSION THREAD

**Discussion by zittert** [patrickvorgers/Home-Assistant-Import-Energy-Data#36] [Q&A] [answered]
Hello Patick,

To start: thanks you all for this project.

Not easy. but after a week of trial and error I had 'result'.
Then in the end nothing showed up in my energyboard.

Some info:

Database(script): SQLite

Sensors commented out except (adapted):
INSERT INTO SENSORS VALUES ('sensor_id_elec_solar',             86,      1000.0,    25.0,            1000.0); /* Change */

Sourcefiles for Enphase:
elec_solar_high_resolution
(CSV colums:)
1665446400	0
1665532801	8298
1665619200	13444
1665705600	20320	
1665792000	32410
1665878400	48152
(etc. 642 rows)

Result:
Execution finished without errors.
Result: query executed successfully. Took 0ms
At line 384:
DROP TABLE IF EXISTS water_low_resolution;

Can you please help me in trhe right direction?

vr.groet, Peter

**Comment by patrickvorgers** [comment]
I am currently on holiday but can you lookup the definition of sensorid 86 in the statistics meta table?

In dbbrowser you can also do a select for sensorid 86 to see whether it contains data.

SELECT * FROM statistics
WHERE
metadata_id = 86

**Reply by zittert** [reply]
Hello Patrick,
Enjoy your holiday! And thanks for answering even now. 
I think you hit the nail because there is no ID 86 found. Probably this is the one i need (from statistics meta table: ID 37(kWh)) sensor.envoy_122220066019_energy_production_today
I'm going to try again (and also try to find out how I got the wrong ID...)
Vr.groet, Peter

**Reply by patrickvorgers** [reply]
I also have Envoy myself. I have used the "Lifetime Energy Production" sensor in my Energy dashboard configuration.
You can lookup the used ID in the energy dashboard configuration.

![image](https://github.com/user-attachments/assets/fb46e1c3-f95d-4c6c-9f3a-915124b5bdd8)

**Reply by zittert** [reply]
Hello Patrick,
You were wright: I used the wrong ID.  ID86 came from "states_meta". Last couple of days tried again with  ID38 sensor.envoy_122220066019_lifetime_energy_production". Result is alike:
Execution finished without errors.
Result: query executed successfully. Took 0ms
At line 384:
DROP TABLE IF EXISTS water_low_resolution;
But again nothing shows up in my energyboard. Can't understand why (Also tried ID37; same result). 
Don't want to give up yet, so when you're back I'd again appreciate a next hint..
vr.groet, Peter

**Reply by patrickvorgers** [reply]
How is your energy dashboard configured? Can you provide a screenshot of the configuration?

**Reply by zittert** [reply]
Hello Patrick, screenshots from a few minutes ago:
![energie1](https://github.com/user-attachments/assets/35040acf-fe01-4fc0-804e-b0ac6547d262)
![energie2](https://github.com/user-attachments/assets/7c95eb5b-dc5e-45b8-b076-c7ae2f4797ed)

**Reply by patrickvorgers** [reply]
> Hello Patrick, screenshots from a few minutes ago:
> 
> ![energie1](https://github.com/user-attachments/assets/35040acf-fe01-4fc0-804e-b0ac6547d262)
> 
> ![energie2](https://github.com/user-attachments/assets/7c95eb5b-dc5e-45b8-b076-c7ae2f4797ed)
> 
> 
> 
> 

For Envoy as solar source you only have to use one source so in your case the "Lifetime production".
You now calculate the solar production twice.

**Reply by zittert** [reply]
I know: just put them in to see if there was a change somewhere.

**Comment by zittert** [comment]
about the screenshots: Because nothing changed, I copied back the database. I will try again to get the historic data in and see if it works. (Assuming I have the right sensors in the e-board...). 
vr.groet, Peter

**Reply by zittert** [reply]
and some remarks:
-enphase website: changing language in right bottom corner is not persistent; had to change it within my account.
-when downloading the data from enphase, it states "Date/Time,**Energy Delivered** (Wh)" instead of "**energy produced**". Changed that in order to get prepare.py to work
-importing the csv-table shows only 20 lines - I guess that's normal
-

**Reply by zittert** [reply]
Screenshots after import:
![energie3](https://github.com/user-attachments/assets/8de5ca09-3bab-48a6-aa90-9c52f7f08db1)
![energie4](https://github.com/user-attachments/assets/90004f04-e11a-410a-9251-bfa54bdfbc54)

But also (still):
![energie5](https://github.com/user-attachments/assets/bbd05304-7080-438b-b360-874bef63e185)

Firsttime install of HA a few weeks ago.

**Comment by patrickvorgers** [comment]
> about the screenshots: Because nothing changed, I copied back the database. I will try again to get the historic data in and see if it works. (Assuming I have the right sensors in the e-board...). 
> 
> vr.groet, Peter 

Can you provide me with the exported Envoy files and the created csv files by the script?

I am back from holiday on the 6th of August and will have a look at it after that.

**Reply by zittert** [reply]
These I used:
[elec_solar_high_resolution.csv](https://github.com/user-attachments/files/16402728/elec_solar_high_resolution.csv)
[3560186_site_energy_production_report.csv](https://github.com/user-attachments/files/16402730/3560186_site_energy_production_report.csv)

**Reply by zittert** [reply]
Also wondering why the "current" and "last seven days" values do not show up in energyboard-config.

**Comment by patrickvorgers** [comment]
Can you make sure that you are using the correct correction values?

Import data is in Wh (Enphase)
Sensor in Home Assistant is in Wh, correct?

In case both are in Wh your configuration should be:
- correction 1,0
- cutoff 25000,0
- invalid 1000000,0

**Reply by zittert** [reply]
Let's see:

- import data state: "Energy Delivered (Wh)"

- sensors in HA database:
![sensors_in_db](https://github.com/user-attachments/assets/8484599e-4312-4606-88a8-007bfe9e28c9)

In the import-sql I used:

INSERT INTO SENSORS VALUES ('sensor_id_elec_solar',             38,      1000.0,    25.0,            1000.0); /* Change */

Tried with ID 38 and 37. These are MWh/kWh. So the INSERT seemed correct to me. But I don't really know  - so these could be values that have to be changed.

**Reply by patrickvorgers** [reply]
Okay I think we got all the information now.

- use sensorid 38
- input data (exported from Envoy) is in Wh (please check)
- sensor data is in MWh
- conversionfactor Wh -> MWH: 1000000 (this value is used to divide the input data by)
- cutoff value in sensor data unit: 0,025 (0,025 MWh / 25 kWh / 25000 Wh)
- invalid value in sensor data unit: 1 (1 MWh / 1000 kWh / 1000000 Wh)

**Reply by zittert** [reply]
Thanks (now i understand the explanation in the .py better)

So tried again with:
INSERT INTO SENSORS VALUES ('sensor_id_elec_solar',             38,      1000000.0,    0.025,            1.0); /* Change */

but afterwards:

![Schermafbeelding 2024-07-30 133305](https://github.com/user-attachments/assets/ca6027d6-9db8-4b07-9745-7afee49ba7da)

![Schermafbeelding 2024-07-30 133553](https://github.com/user-attachments/assets/5273c903-67cf-4eca-bdd3-5cb02214c0e3)

![Schermafbeelding 2024-07-30 133629](https://github.com/user-attachments/assets/1bd78e1c-11b4-4b24-bd87-8dbb1dfdcc85)

nothing before 12th of july.

Either I'm still doing something wrong or the project changes are not written to the database.
After clicking 'write changes' the buttons stay grayed out - is it supposed to be like that? 

Patrick, please enjoy your holiday. After your return I would be thankful to hear from you again.

**Reply by patrickvorgers** [reply]
I will 😃
I think you have the wrong sensor now in your energy dashboard. We are adding the data to the lifetime sensor (ID 38) and in your energy dashboard I see the today sensor.

![image](https://github.com/user-attachments/assets/3fe3ba87-8733-467e-b022-a622a829fc8e)

You could also verify whether the data is in the database by creating a temporary dashboard in Home Assistant with something like a entity/entities card and than select the entity and select "show more", there you can select your own date range.

**Reply by zittert** [reply]
It is in there (14 envoy...) but probably not in the right place then. I'll try again (because restored the backup).

**Reply by zittert** [reply]
same same...

![Schermafbeelding 2024-07-30 143935](https://github.com/user-attachments/assets/7c5c4535-280b-409a-91e3-a2cb7ea7b287)
![Schermafbeelding 2024-07-30 143958](https://github.com/user-attachments/assets/364048e1-25ba-4076-b138-eb3086617914)

**Reply by zittert** [reply]
same same ...

![Schermafbeelding 2024-07-30 143935](https://github.com/user-attachments/assets/2158398a-f5d6-4ca5-be92-1776b00118c5)
![Schermafbeelding 2024-07-30 143958](https://github.com/user-attachments/assets/9d658818-3f28-482d-bf7e-444abba278e9)

**Reply by zittert** [reply]
sorry doubled it..

**Reply by zittert** [reply]
I'll restore and beginning to think the import csv can be wrong because I had troubles with that (enphase -> prepare.py -> csv)

**Comment by patrickvorgers** [comment]
@zittert Did you manage to get the issue resolved?

If not can you provide me with the following information so that I can try to reproduce the issue:
- Home Assistant database
- Enphase export and produced csv file
- Adapted script

**Reply by zittert** [reply]
Hello Patrick,
I did not succeed. (Also noticed new issue/discussion with probably the same subject)
Already thanks for trying to reproduce. But wondering if it is safe to send my database in here?
vr.groet, Peter

**Reply by patrickvorgers** [reply]
You are right 
I think you can look me up on the Home Assistant community forum and send me a private message there.

**Reply by zittert** [reply]
Patrick, I found "lasoul" in there but cannot figure out how to send you a private message.

**Reply by patrickvorgers** [reply]
On the "Home Assistant Community"

- Logon to your account
- Click on your profile icon (top right corner)
- In the dropdown select the messages icon (message envelope)
- Click the expand button (Arrow down button)
- You are now in the messages overview. Create a new message.

**Comment by patrickvorgers** [comment]
Thanks for sharing the database. I tested the provided information and your configuration is correct.
When I run the script it imports the data into ID 38 and when I connect your database to my test instance it shows the historical data.
Did you make sure that everytime you run the script you have to import the CSV file again? Because the temporay table is deleted after each run of the script!

I had to configure ```sensor.envoy_122220066019_lifetime_energy_production``` as the solar production.

![2023](https://github.com/user-attachments/assets/93fb9dcc-c5d2-49d5-a727-47f546069eed)
![2024](https://github.com/user-attachments/assets/e8b7bf7d-f9a7-4980-bb98-29cb961c79ee)
![Energie configuratie](https://github.com/user-attachments/assets/4a1ab709-8d97-4dd5-8073-4a2a9db66184)

**Reply by zittert** [reply]
Hi Patrick,
"everytime you run the script you have to import the CSV file again".  I followed this instruction to the point:
[SQL howto.pdf](https://github.com/user-attachments/files/16645778/SQL.howto.pdf)
If it worked in your test-instance, it must work in my HA. I'm going to run it again. If no result, I'm (still) doing something wrong...
Thanks again! Peter

**Reply by Stephan-4711** [reply]
A user search for the mistake on his side, you a very seldom species 😅 #justAJoke

**Reply by patrickvorgers** [reply]
> Hi Patrick, "everytime you run the script you have to import the CSV file again". I followed this instruction to the point: [SQL howto.pdf](https://github.com/user-attachments/files/16645778/SQL.howto.pdf) If it worked in your test-instance, it must work in my HA. I'm going to run it again. If no result, I'm (still) doing something wrong... Thanks again! Peter

Okay, hopefully it works for you as well!

**Reply by Stephan-4711** [reply]
Ähm... In 2024.8 they changed the Database and wrote they optimized stuff, maybe this is related to the problem?

**Reply by patrickvorgers** [reply]
My 

> Ähm... In 2024.8 they changed the Database and wrote they optimized stuff, maybe this is related to the problem?

From what I have read the changes were on MariaDB and not on SQLite. They changed the record ID's to be BIGINT.
The test that I did with the provided database was on my HA 2024.8.2 test instance.

**Reply by zittert** [reply]
Hi Patrick, hi Stephan, todays update:
I tried again with no result. Then I uploaded the (my) database that Patrick send back, and the history appeared. So I must be doing something wrong. Few minutes ago I restored my backup from yesterday (the one send back is missing recent two weeks). I'll keep trying... Greetings, Peter

**Reply by patrickvorgers** [reply]
@zittert, if you want I can run the script against your latest backup (now). That way you have an updated database and you don't have to run the script yourself. For me it is just a few minutes work. I can do that directly if you want.

**Reply by patrickvorgers** [reply]
> Hi Patrick, hi Stephan, todays update: I tried again with no result. Then I uploaded the (my) database that Patrick send back, and the history appeared. So I must be doing something wrong. Few minutes ago I restored my backup from yesterday (the one send back is missing recent two weeks). I'll keep trying... Greetings, Peter

Maybe stating the obvious but I don't know why it is not working for you: After the script has completed succesfully, make sure to press "write changes". This takes some while to complete (button is grayed out after it is completed)

**Reply by zittert** [reply]
Hi Patrick,
Thanks, that would be nice but I'm going to try myself first. Its obvious I'm repeating some mistake, and would like to find out at what point. I suspect when importing the csv. I'll let you know (again).

**Reply by zittert** [reply]
and also trying your new script..

**Reply by patrickvorgers** [reply, marked as answer]
Okay,....
Things to verify

- After import of the CSV file check the contents of the ```elec_solar_high_resolution``` table. Go to ```Browse Data``` and select the ```elec_solar_high_resolution``` table. With the CSV file you provided me the table contains 615 records (bottom left corner)
- After you have run the script check the contents of the ```statistics``` table for ```metadata_id``` 38. Go to ```Browse Data``` and select the ```statistics``` table. Under the columnname ```metadata_id``` fill in 38 in the ```filter``` edit box. This will filter the data so that it only displays the data for sensor_id 38. Press on the ```start_ts``` label so that the data is sorted from low to high. With the provided CSV data you first record should have a ```start_ts```  of 1665532800.0. In case the import did not work the first record is probably in the range of 1720198800.0.

**Reply by zittert** [reply]
Thanks for explaining. And: the new (battery).py worked great.

**Reply by zittert** [reply]
Patrick, indeed I get start_ts 1720198800.0. Can you think of any cause? Like working in windows11 instead of linux?
Don't like to give up but probably best to accept your offer to run it for me... Peter

**Reply by zittert** [reply]
Patrick, thanks for all your help solving my problem and pointing out my mistake. Turned out that - among other things - the sql_engine version is important.
